galleries.py

```
@@ -19,8 +24,33 @@
 
+def get_final_NASA_url(url):
+    """ Since NASA is now redirecting all the original photojournal links, 
+        this will (hopefully) get the redirected links for the updated galleries
+        
+    Input:
+        url     Original photojournal URL w/PIA embedded, example:
+                https://photojournal.jpl.nasa.gov/catalog/PIA24615
+    """
+    try:
+        opener = urllib.request.build_opener()
+        request = urllib.request.Request(url)
```

Ruff (0.14.14)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)